### PR TITLE
core/thread: move thread configuration into seperate header

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This PR moves some thread configuration defines from thread.h into thread_config.h.
As is, this only moves code around, but it is the first step to untangle the thread.h API from cpu specific stuff (see e.g., [here](https://github.com/RIOT-OS/RIOT/pull/15595#issuecomment-762094819)).
It is also a necessary step for providing alternative scheduler / core implementations.

### Testing procedure

This only moves code around, so a CI success should be fine.

### Issues/PRs references

Suggested in #15595.

### CODE REVIEW

File: core/include/thread_config.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ *
+ * @file
+ * @brief       Thread configuration defines
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef THREAD_CONFIG_H
+#define THREAD_CONFIG_H
+
```
Comment (maribu) [inline core/include/thread_config.h:24]: The CI wants the usual C++ compatibility boilerplate.

**Review by maribu — COMMENTED** [review]
The CI has some feedback.

A follow up step would IMO be to move out the arch specific configs to `cpu/<SOME_ARCH>_common/include/thread_config_arch.h`, so that `cpu_conf.h` is no longer needed.

**Comment by maribu** [comment]
@kaspar030: Ping? I volunteer to do the follow up. The vendor files get included has again bitten me in the ass, I finally want to get rid of this issue.

File: core/include/thread_config.h (hunk 1 above)
Comment (kaspar030) [reply core/include/thread_config.h:24]: fixed!

**Review by maribu — APPROVED** [review]
lgtm, please squash

**Comment by kaspar030** [comment]
squashed

**Pull request merged**